test/metabase/query_processor_test/string_extracts_test.clj

```diff
@@ -0,0 +1,60 @@
+(ns metabase.query-processor-test.string-extracts-test
+  (:require [metabase.query-processor-test :refer :all]
+            [metabase.test.data :as data]
+            [metabase.test.data.datasets :as datasets]))
+
+(defn- test-string-extract
+  [expr]
+  (->> {:expressions {"test" expr}
+        :fields      [[:expression "test"]]
+        ;; To ensure stable ordering
+        :order-by    [[:asc [:field-id (data/id :venues :id)]]]
+        :limit       1}
+       (data/run-mbql-query venues)
+       rows
+       ffirst))
+
+(datasets/expect-with-drivers (non-timeseries-drivers-with-feature :expressions)
```

Comment:
I made a composite namespace called `metabase.test` that has 95% of all the utility functions you need to write Metabase tests all in a single place. So you can use that instead of the combo of `data`/`datasets`/etc.

Like my previous comment, you shouldn't use `expect-with-drivers` for any new tests.

I've deprecated `non-timeseries-drivers` a while back. It's not an accurate name since it doesn't include Google Analytics. It is replaced by `normal-drivers`

This is the how I would write test sorts of tests using the new style:

```clj
(require '[metabase.test :as mt])

(deftest my-test
  (mt/test-drivers (mt/normal-drivers-with-feature :expressions)
    (is (= expected
           (mt/run-mbql-query ...)))))
```